src/app/password-reset-page/password-reset-page.component.ts

```diff
@@ -0,0 +1,106 @@
+import {Component, OnInit} from '@angular/core';
+import {AccountService} from '../../services/account.service';
+import {NgbModal, NgbModalRef} from '@ng-bootstrap/ng-bootstrap';
+import {ModalContentComponent} from '../modal-content/modal-content.component';
+import {ActivatedRoute, Params, Router} from '@angular/router';
+
+@Component({
+  selector: 'app-account-recovery-page',
+  templateUrl: './password-reset-page.component.html',
+  styleUrls: ['./password-reset-page.component.css']
+})
+export class PasswordResetPageComponent implements OnInit {
+
+  private minPasswordLength: number;
+  private isLoggedIn: boolean;
+  private submitDisabled: boolean;
+
+  private password: string;
+  private confirmedPassword: string;
+  private showErrorPassword: boolean;
+  private showErrorConfirmedPassword: boolean;
+
+  constructor(
+    private accountService: AccountService,
+    private router: Router,
+    private activatedRoute: ActivatedRoute,
+    private modalService: NgbModal
+  ) {}
+
+  public ngOnInit() {
+    this.minPasswordLength = AccountService.getMinPasswordLength();
+    this.isLoggedIn = false;
+    this.submitDisabled = false;
+    this.showErrorPassword = false;
+    this.showErrorConfirmedPassword = false;
```

Comment:
I know this is code that already exists, but for the future, all of this variable initialization can be done above where you're declaring the variables. I think the result is exactly the same, but you have the variables in fewer places and the onInit function is more focused.

Reply:
That's true. I guess I'm just used to this style of code.

Reply:
One reason I prefer this style of code where the variable is initialized in the constructor (or in this case in the on init method), is because it makes it very clear that these are just initial values, and better communicates that the value of the variable could be anything.

Reply:
However, you are right in that it does clutter up the code a bit.